html/mycakeapp/src/Controller/ReservesController.php

```diff
@@ -33,32 +34,76 @@ public function initialize()
                     'record_number' => $_SESSION['seat']['record_number'],
                 ])
                 ->toArray();
-            $seatReservation[0]['is_cancelled'] = '1';
-            if (!($this->SeatReservations->save($seatReservation['0']))) {
+            $ExReservation[0]['is_cancelled'] = '1';
+            if (!($this->SeatReservations->save($ExReservation['0']))) {
                 $this->request->session()->delete('seat');
                 return $this->redirect(['controller' => 'error']);
             }
             $this->request->session()->delete('seat');
         }
+        //席予約情報を取得
+        $seatReservations = $this->SeatReservations->find()
+            ->where(['schedule_id' => $_SESSION['schedule_id']])
+            ->andWhere(['is_cancelled' => 0]);
+        foreach ($seatReservations as $seatReservation) {
+            $Reserved[] = $seatReservation['column_number'] . '-' . $seatReservation['record_number'];
+        }
+        //予約済み情報を取得
+        $Reservations = $this->ReservationDetails->find()
+            ->where(['schedule_id' => $_SESSION['schedule_id']])
+            ->andWhere(['is_cancelled' => 0]);
+        foreach ($Reservations as $Reservation) {
+            $Reserved[] = $Reservation['column_number'] . '-' . $Reservation['record_number'];
+        }
+        if (isset($Reserved)) {
+            $this->set(compact('Reserved'));
+        }
+        if ($this->request->isPost()) {
+            $seatNumber = explode("-", $_POST['seat']);
+            if (in_array($_POST['seat'], $Reserved)) {
+                return $this->redirect(['controller' => 'error']);
+            }
+            $entity = $this->SeatReservations->newEntity();
+            $entity['member_id'] = $this->Auth->user('id');
+            $entity['schedule_id'] = $_SESSION['schedule_id'];
+            $entity['column_number'] = $seatNumber[0];
+            $entity['record_number'] = $seatNumber[1];
+            if ($this->SeatReservations->save($entity)) {
+                $_SESSION['seat']['member_id'] = $this->Auth->user('id');
+                $_SESSION['seat']['schedule_id'] = $_SESSION['schedule_id'];
+                $_SESSION['seat']['column_number'] = $seatNumber[0];
+                $_SESSION['seat']['record_number'] = $seatNumber[1];
+                $this->redirect(['action' => 'ticket']);
+            }
+            $this->redirect(['controller' => 'error']);
+        }
+        //シアターファイル名を取得
+        $seatDetails = $this->Schedules->get($_SESSION['schedule_id'], [
+            'contain' => ['Movies', 'Theaters']
+        ]);
+        $this->viewBuilder()->setLayout('frame-title');
+        $title = '座席指定';
+        $week = ['日', '月', '火', '水', '木', '金', '土'];
+        $startdate = $seatDetails['start_date']->format('m月d日　G:i') . '(' . $week[$seatDetails['start_date']->format('w')] . ')';
+        $finishdate = date('G:i', strtotime('+' . $seatDetails['movie']['screening_time'] . 'minute', strtotime($startdate)));
```

Comment:
終了時刻がずれているので修正お願いします🙇‍♂️🙇‍♂️
(チケット選択ページの終了時刻と比較すると分かりやすいと思います!)
```suggestion
        $finishdate = date('G:i', strtotime('+' . $seatDetails['movie']['screening_time'] . 'minute', strtotime($seatDetails['start_date'])));
```